Source/AssetRipper.Processing/AnimatorControllers/VirtualAnimationFactory.cs

```diff
@@ -159,207 +160,446 @@ private static IChildMotion AddAndInitializeNewChild(this IBlendTree tree, Proce
 		}
 
 		private static void CreateEntryTransitions(
-			IAnimatorStateMachine generatedStateMachine,
+			IAnimatorStateMachine stateMachine,
 			IStateMachineConstant stateMachineConstant,
 			ProcessedAssetCollection file,
-			uint ID,
+			uint FullPathID,
 			IReadOnlyList<IAnimatorState> States,
 			AssetDictionary<uint, Utf8String> TOS)
 		{
-			if (generatedStateMachine.Has_EntryTransitions() && stateMachineConstant.Has_SelectorStateConstantArray())
+			if (stateMachine.Has_EntryTransitions() && stateMachineConstant.Has_SelectorStateConstantArray())
 			{
-				foreach (OffsetPtr_SelectorStateConstant selectorPtr in stateMachineConstant.SelectorStateConstantArray)
+				foreach (SelectorStateConstant selector in stateMachineConstant.SelectorStateConstantArray)
 				{
-					SelectorStateConstant selector = selectorPtr;
-					if (selector.FullPathID == ID && selector.IsEntry)
+					if (selector.IsEntry && selector.FullPathID == FullPathID)
 					{
 						for (int i = 0; i < selector.TransitionConstantArray.Count - 1; i++)
 						{
 							SelectorTransitionConstant selectorTrans = selector.TransitionConstantArray[i].Data;
-							IAnimatorTransition transition = CreateAnimatorTransition(file, stateMachineConstant, States, TOS, selectorTrans);
-							generatedStateMachine.EntryTransitionsP.Add(transition);
+							// Entries only point to States; SubStateMachine[] and IAnimatorStateMachine can be null
+							IAnimatorTransition transition = CreateAnimatorTransition(file, stateMachineConstant, States, null, null, TOS, selectorTrans);
+							stateMachine.EntryTransitionsP.Add(transition);
 						}
+
+						// Default State
+						int defaultStateIdx = (int)selector.TransitionConstantArray[^1].Data.Destination;
+						IAnimatorState defaultState = States[defaultStateIdx];
+						stateMachine.DefaultState.CopyValues(defaultState, new PPtrConverter(stateMachine));
```

Comment:
This line is suspicious. I'm actually surprised that it compiled. The first parameter should be a PPtr.

Reply:
However, you probably want `DefaultStateP`.

Reply:
fixed